### DISCUSSION TITLE
Seeking Clarity on REALITY and VMess and GFW

### DISCUSSION THREAD

**Discussion by houmie** [XTLS/Xray-core#2233] [Q&A]
I'm hoping to get some help with a few things:

1) I'm looking at examples like this one: https://github.com/lxhao61/integrated-examples/tree/main/Xray(E%2BB%2BD%2BG%2BA)%2BCaddy(N). These examples use different protocols and have fallback options to a decoy website. My worry is about less secure protocols. If we use them, the GFW in Iran or China might block the whole server's IP. So, it might not be a good idea to have servers based on REALITY and less secure protocols on the same VPS. Does this make sense to you?

2) It seems to me that using VLESS+VISION+TCP+REALITY is the most secure. But some people prefer VLESS+H2+REALITY. This second choice has a problem, though. It might make it easier to detect because of the TLS-in-TLS issue. Am I right about this? If yes, what's good about this second choice?

3) I'm wondering why VMess+WebSocket+TLS is still popular. There was a report in 2020 saying that the GFW has compromised VMess: https://gfw.report/blog/v2ray_weaknesses/en/. Wouldn't it be safer to use VLESS+WebSocket+TLS?

Thank you for your help.

**Comment by lhbdhr** [comment]
因为clash不支持vless，又不喜欢torjan的话，就只能vmess
